### ISSUE TITLE
Add engineering process

### ISSUE THREAD

**Issue by jenstroeger** [sphinx-notes/sphinx-markdown-builder#4]
Based on the [python-package-template](https://github.com/jenstroeger/python-package-template) repo I think it would make sense to bring over some of the engineering and automation into this repo: [pylint](http://pylint.pycqa.org/) and [black](https://github.com/psf/black) at least.

Given the number of issues we’re opening we should also replace the [examples/javascript](https://github.com/sphinx-notes/markdown-builder/tree/master/examples/javascript) folder with proper [pytest](https://github.com/pytest-dev/pytest)?

**Comment by SilverRainZ** [comment]
Agree. I can not run examples/javascript locally due to the old dependency.

**Comment by SilverRainZ** [comment]
Do you have time to work on this? I am fixing #2 and then  #1.

**Comment by jenstroeger** [comment]
I’ll schedule it for Tuesday. Also have another engineer who will pick up some work now.

**Comment by jenstroeger** [comment]
@lecram you’re familiar with the [template](https://github.com/jenstroeger/python-package-template/) already. I’ll try to take a stab this weekend, and will ping you if not.

@SilverRainZ let’s start with an empty pytest folder and add unit tests as we go. This PR, I think, should be concerned with putting things into place instead of adding tests?

**Comment by SilverRainZ** [comment]
I think we should need at least one test to show pytest actually works.

We can test against the changes we made, such as #5 and #6.

**Comment by SilverRainZ** [comment]
Ping @jenstroeger

**Comment by jenstroeger** [comment]
Ugh my apologies, it’s been very busy the past few weeks. We’re currently fixing an issue in our [Py package template](https://github.com/jenstroeger/python-package-template) after which we’ll have a good baseline to merge into this repo. Are you ok waiting a week or so?

**Comment by SilverRainZ** [comment]
No problem, just a reminder.

**Comment by jenstroeger** [comment]
@SilverRainZ let’s perhaps do this in several steps:

1. replace [legacy `setup.py`](https://pip.pypa.io/en/stable/reference/build-system/setup-py/) with the recommended [pyproject.toml](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/); then
2. add support for [git hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) and use [pre-commit](https://pre-commit.com/) to do add a few simple hooks (lint, flake, upgrade, [see more](https://github.com/jenstroeger/python-package-template/blob/main/.pre-commit-config.yaml)); then
3. change and clean up testing; then
4. expand on the [Community files](https://github.com/sphinx-notes/markdown-builder/community); then
5. add a few useful [Github Actions](https://github.com/features/actions) (see [examples](https://github.com/jenstroeger/python-package-template/tree/main/.github)).

We can adjust the Makefile as we move forward, and add more flexibility to it (see [example](https://github.com/jenstroeger/python-package-template/blob/main/Makefile)).

Considering that [Sphinx itself requires at least Python 3.6](https://github.com/sphinx-doc/sphinx/blob/5.x/pyproject.toml#L16), I think the same should apply here. The metadata classifiers seem outdated

https://github.com/sphinx-notes/markdown-builder/blob/b9f424b5e5001a8192e1b91d1b29afd60772bca7/setup.py#L32-L37

and the Makefile seems to expect Python 3. This can be fixed as part of step 1. above.

**Comment by SilverRainZ** [comment]
@jenstroeger That sounds good :D 

I am quite busy recently so maybe can't help :'(
